src/lib/middleware/response-time-metrics.ts

```diff
@@ -2,21 +2,28 @@ import * as responseTime from 'response-time';
 import EventEmitter from 'events';
 import { REQUEST_TIME } from '../metric-events';
 import { IFlagResolver } from '../types/experimental';
+import { InstanceStatsService } from 'lib/services';
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 const _responseTime = responseTime.default;
 
+const appNameReportingThreshold = 100;
+
 export function responseTimeMetrics(
     eventBus: EventEmitter,
     flagResolver: IFlagResolver,
+    instanceStatsService: Pick<InstanceStatsService, 'getAppCountSnapshot'>,
 ): any {
     return _responseTime((req, res, time) => {
         const { statusCode } = res;
-
         const pathname = req.route ? req.baseUrl + req.route.path : '(hidden)';
 
         let appName;
-        if (flagResolver.isEnabled('responseTimeWithAppName')) {
+        if (
+            flagResolver.isEnabled('responseTimeWithAppName') &&
+            (instanceStatsService.getAppCountSnapshot('7d') ||
+                appNameReportingThreshold) < appNameReportingThreshold
```

Comment:
If app count was undefined, fallback to `appNameReportingThreshold`. This prevents us from reporting app names until we have the data which might protect us from some edge cases

Reply:
if flagresolver is false, will it evaluate the stuff after &&?

Reply:
what will appname be if we have this disabled or too many apps?

Reply:
if flagresolver is false it won't resolve the other part of the `&&`, it's a short-circuit and ;)

> what will appname be if we have this disabled or too many apps?

It won't have appname, which will translate to `"undefined"` (as the current behavior).

Example from prometheus metrics:
```
# HELP http_request_duration_milliseconds App response time
# TYPE http_request_duration_milliseconds summary
http_request_duration_milliseconds{quantile="0.1",path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds{quantile="0.5",path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds{quantile="0.9",path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds{quantile="0.95",path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds{quantile="0.99",path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds_sum{path="/*",method="GET",status="304",appName="undefined"} 8.32329
http_request_duration_milliseconds_count{path="/*",method="GET",status="304",appName="undefined"} 1
http_request_duration_milliseconds{quantile="0.1",path="(hidden)",method="GET",status="200",appName="undefined"} 8.339421
http_request_duration_milliseconds{quantile="0.5",path="(hidden)",method="GET",status="200",appName="undefined"} 9.418512
http_request_duration_milliseconds{quantile="0.9",path="(hidden)",method="GET",status="200",appName="undefined"} 10.466320999999999
http_request_duration_milliseconds{quantile="0.95",path="(hidden)",method="GET",status="200",appName="undefined"} 10.466320999999999
http_request_duration_milliseconds{quantile="0.99",path="(hidden)",method="GET",status="200",appName="undefined"} 10.466320999999999
http_request_duration_milliseconds_sum{path="(hidden)",method="GET",status="200",appName="undefined"} 28.224254000000002
http_request_duration_milliseconds_count{path="(hidden)",method="GET",status="200",appName="undefined"} 3
http_request_duration_milliseconds{quantile="0.1",path="(hidden)",method="GET",status="401",appName="undefined"} 1.847349
http_request_duration_milliseconds{quantile="0.5",path="(hidden)",method="GET",status="401",appName="undefined"} 4.5549409999999995
http_request_duration_milliseconds{quantile="0.9",path="(hidden)",method="GET",status="401",appName="undefined"} 11.237288
http_request_duration_milliseconds{quantile="0.95",path="(hidden)",method="GET",status="401",appName="undefined"} 11.237288
http_request_duration_milliseconds{quantile="0.99",path="(hidden)",method="GET",status="401",appName="undefined"} 11.237288
http_request_duration_milliseconds_sum{path="(hidden)",method="GET",status="401",appName="undefined"} 27.669517999999997
http_request_duration_milliseconds_count{path="(hidden)",method="GET",status="401",appName="undefined"} 5
```